File: 1078-remove-outermost-parentheses/1078-remove-outermost-parentheses.py
class Solution:
    def removeOuterParentheses(self, s: str) -> str:
        result = []
        opened = 0
        for i in s:
            if i == "(" and opened > 0:
                result.append(i)
            if i == ")" and opened > 1:
                result.append(i)
            opened += 1 if i == "(" else -1
        return "".join(result)
        

File: 1078-remove-outermost-parentheses/1078-remove-outermost-parentheses.cpp
class Solution {
public:
    string removeOuterParentheses(string s) {
        int count = 0;
        string ans = "";
        for(int i=0;i<s.size();i++){
            if(s[i] == '('){
                if(count>0){
                    ans += '(';
                }
                count++;
            }
            if(s[i] == ')'){
                if(count>1){
                    ans+=')';
                }
                count--;
            }
        }
        return ans;

    }
};